src/main/java/com/vadmack/mongodbtest/security/JwtTokenUtil.java

```diff
@@ -0,0 +1,56 @@
+package com.vadmack.mongodbtest.security;
+
+import com.vadmack.mongodbtest.entity.User;
+import io.jsonwebtoken.*;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+@RequiredArgsConstructor
+public class JwtTokenUtil {
+
+    private final Logger logger;
+
+    @Value("${authorization.secret}")
+    private String secret;
+
+    public String generateAccessToken(User user) {
+        return Jwts.builder()
+                .setSubject(String.format("%s,%s", user.getId(), user.getUsername()))
+                .setIssuedAt(new Date())
+                .setExpiration(new Date(System.currentTimeMillis() + 24 * 60 * 60 * 1000)) // 24 hours
+                .signWith(SignatureAlgorithm.HS512, secret)
+                .compact();
+    }
+
+    public boolean validate(String token) {
+        try {
+            Jwts.parser().setSigningKey(secret).parseClaimsJws(token);
+            return true;
+        } catch (SignatureException ex) {
+            logger.error("Invalid JWT signature - {}", ex.getMessage());
+        } catch (MalformedJwtException ex) {
+            logger.error("Invalid JWT token - {}", ex.getMessage());
+        } catch (ExpiredJwtException ex) {
```

Comment:
You can combine `catch (MalformedJwtException ex)` and `catch (ExpiredJwtException ex)` like this:
`catch (MalformedJwtException | ExpiredJwtException  ex)`
because you process them similarly